tc/core/cuda/cuda_libraries.h

```diff
@@ -43,8 +43,8 @@ constexpr auto defines = R"C(
 
 constexpr auto warpSyncFunctions = R"C(
 // Before CUDA 9, syncwarp is a noop since warps are always synchronized.
-#if __CUDACC_VER_MAJOR__ < 9
-__device__ void __syncwarp(unsigned mask = 0xFFFFFFFF) {}
+#if defined(__CUDACC_VER_MAJOR__) && __CUDACC_VER_MAJOR__ < 9
+inline __device__ void __syncwarp(unsigned mask = 0xFFFFFFFF) {}
```

Comment:
This looks fishy.  Why can you assume CUDA 9 when `__CUDACC_VER_MAJOR__` is not defined?

Reply:
good catch thanks!